src/surface.rs

```diff
@@ -47,6 +47,60 @@ impl Surface {
                                                                   height))
         }
     }
+
+    pub fn get_mime_data(&self, mime_type: &str) -> Option<Vec<u8>> {
+        let mut data_ptr: *mut u8 = ptr::null_mut();
+        let mut length = 0u64;
+        unsafe {
+            ffi::cairo_surface_get_mime_data(
+                self.to_raw_none(),
+                mime_type.to_glib_none().0,
+                &mut data_ptr,
+                &mut length,
+            );
+            if !data_ptr.is_null() && length != 0 {
+                Some(slice::from_raw_parts(
+                    data_ptr as *const u8,
+                    length as usize,
+                ).to_vec())
+            } else {
+                None
+            }
+        }
+    }
+
+    pub fn set_mime_data<T: AsRef<[u8]> + Send + 'static>(&self,
+                                                          mime_type: &str,
+                                                          slice: T) -> Result<(), Status> {
```

Comment:
This one should probably be indented a bit differntly:

```rust
    pub fn set_mime_data<T: AsRef<[u8]> + Send + 'static>(
        &self,
        mime_type: &str,
        slice: T) -> Result<(), Status>
```

Or something like that. Indent logically, not visually :)